src/analytics/tests/test_pipelines.py

```diff
@@ -198,13 +211,63 @@ def test_create_concept_drift_pipeline_drift_detected(self):
+            df_multi,
+            "target",
         )
         eval_score = eval["precision"]
         assert (round(eval_score, 2)) == 0.97
+
+    def test_create_xai_pipeline_classification_per_inference_row(self):
+        binary_class_report1 = create_xai_pipeline_classification_per_inference_row(
+            df_binary, "target", df_binary_inference_row1, "binary"
+        )
+        multi_class_report1 = create_xai_pipeline_classification_per_inference_row(
+            df_multi, "target", df_multi_inference_row1, "multi_class"
+        )
+        binary_class_report2 = create_xai_pipeline_classification_per_inference_row(
+            df_binary, "target", df_binary_inference_row2, "binary"
+        )
+        multi_class_report2 = create_xai_pipeline_classification_per_inference_row(
+            df_multi, "target", df_multi_inference_row2, "multi_class"
+        )
+
+        binary_contribution_check_one = binary_class_report1["worst perimeter"]
+        binary_contribution_check_two = binary_class_report2["worst texture"]
+        multi_contribution_check_one = multi_class_report1["hue"]
+        multi_contribution_check_two = multi_class_report2["alcohol"]
+
+        # We delete the models and the directory created by the create models for classification pipelines
+        os.remove(f"{test_model_path}/lgb_binary.pkl")
+        os.remove(f"{test_model_path}/lgb_multi.pkl")
+        os.rmdir(test_model_path)
+
+        assert (round(binary_contribution_check_one, 3)) == -0.464
+        assert (round(binary_contribution_check_two, 1)) == -0.1
+        assert (round(multi_contribution_check_one, 2)) == -0.09
+        assert (round(multi_contribution_check_two, 3)) == 0.076
+
+    # The below unit tests are not activate for now. Performs testing on explainability pipeline per inference dataset.
+    # def test_create_xai_pipeline_classification_per_inference_dataset(self):
+    #     binary_class_report = create_xai_pipeline_classification(df_binary,"target",df_binary_inference,"binary_classification")
+    #     multi_class_report = create_xai_pipeline_classification(df_multi,"target",df_multi_inference,"multiclass_classification")
+
+    #     binary_contribution_check_one = binary_class_report["row0"]["worst perimeter"]
+    #     binary_contribution_check_two = binary_class_report["row2"]['worst texture']
+    #     multi_contribution_check_one = multi_class_report["row0"]["hue"]
+    #     multi_contribution_check_two = multi_class_report["row9"]["proanthocyanins"]
+
+    #     assert (len(binary_class_report)) == len(df_binary_inference)
+    #     assert (len(multi_class_report)) == len(df_multi_inference)
+
+    #     assert (round(binary_contribution_check_one, 3)) == 0.253
+    #     assert (round(binary_contribution_check_two, 2)) == -0.09
+    #     assert (round(multi_contribution_check_one, 2)) == -0.08
+    #     assert (round(multi_contribution_check_two, 3)) == -0.023
```

Comment:
I suggest we remove these comments if we dont need them

Reply:
Actually there is a use for those lines. But it is a future one. In any case, if we have an efficient way of deleting them but not rewrite them from scratch in the future, go on.